### DISCUSSION TITLE
GitHub action still waiting for a runner after near 23h

### DISCUSSION THREAD

**Discussion by cecilios** [community/community#23531] [New to GitHub]
<p>I’m trying to replace Travis CI and this is my first attempt to use GitHub Actions. My first test worked perfectly but this second test has been in queue waiting for a runner for more than 22h and still waiting. Is this normal? What is the expected average delay for running an Action? Is there some problem with my Action?</p>

**Comment by airtower-luna** [comment]
<p>That sounds unusual. If there was an error in your workflow you should get an error message, not an endless wait. The only where I would expect a practically endless wait is if you’re using self-hosted runners and those are offline. :sweat_smile:</p>
<p>Otherwise, you can check if there’s a known issue at <a href="https://www.githubstatus.com/">https://www.githubstatus.com/</a>, and possibly file a bug report: <a href="https://support.github.com/contact/bug-report" class="inline-onebox">Bug report - GitHub Support</a></p>

**Comment by cecilios** [comment]
<p>Thank you. I think I do not use a self-hosted runner (it is a simple action to trigger a build after each PR, but … I’m new to this). The action is still waiting (1d 43m) so I will  open a bug report. Thank you again.</p>

**Comment by jsoref** [comment]
<p>Occasionally things get lost. You should treat things as dead if they aren’t resolved w/in 3 days. At that point, you can file a ticket to <a href="http://support.github.com">support.github.com</a> w/ a link to the run and they can clean it out. (Typically these are because of some GHA outage for which <a href="http://githubstatus.com">githubstatus.com</a> would have reported an item, although that isn’t always the case.)</p>
<p>Recently I found a pair that got stuck ~8 months ago.</p>
<p>Overall, for the fairly large number of runs we’ve had, the handful of items that were lost.</p>

**Comment by cecilios** [comment, marked as answer]
<p>Thank you. I didn’t see your post until today.</p>
<p>The problem was mine. I that I coded:</p>
<blockquote>
<pre><code>runs-on: ${{matrix.os}}
</code></pre>
</blockquote>
<p>instead of</p>
<blockquote>
<pre><code>runs-on: ${{matrix.os}}-latest
</code></pre>
</blockquote>
<p>And therefore there were not runners available with that name, and the job was endlessly waiting. Perhaps GitHub should detect requests for non existing runners and abort with a meaningful message error.</p>
